modules/statistics/statistics.js

```diff
@@ -238,10 +239,41 @@ Statistics.prototype.startRemoteStats = function(peerconnection) {
 
 Statistics.localStats = [];
 
-Statistics.startLocalStats = function(stream, callback) {
+Statistics.startLocalStats = function(track, callback) {
+    track.addEventListener(
+        JitsiTrackEvents.LOCAL_TRACK_STOPPED,
+        () => {
+            Statistics.stopLocalStats(track);
+        });
+
+    track.addEventListener(
```

Comment:
I understand that you need to kill the context and that's why we need to stop all localStats instances but the part I don't like is that there is only one localStats instance related to the track that triggers the no data from source event. Anyway I don't think there is anything you can do about it... and also I'm not sure if we will actually have a scenario with multiple local audio tracks.  

Could you please add a comment in the code with explanation what we are actually fixing here? 

Also since this is an issue only with mobile Safari could you please add the listener only for mobile Safari?

Reply:
> Also since this is an issue only with mobile Safari could you please add the listener only for mobile Safari?

Good point, we should do that. For any iOS browser actually.

Chrome on Android doesn't seem to emit the muted events for local tracks so it doesn't apply there...

Reply:
Did for any iOS browser